example/simple_filter.cpp

```diff
@@ -0,0 +1,38 @@
+//include libraries for image io, kernel definition and convolution
+#include <boost/gil/image.hpp>
+#include <boost/gil/typedefs.hpp>
+#include <boost/gil/extension/io/jpeg.hpp>
+#include <boost/gil/extension/numeric/kernel.hpp>
+#include <boost/gil/extension/numeric/convolve.hpp>
+
+
+int main(int argc,char *argv[]) {
+    using namespace boost::gil;
+    if(argc!=2){
+        printf("Usage: simple_filter <image_path>");
+        return 0;
+    }
+    rgb8_image_t img;
+    //read image
+    read_image(*argv[1], img, jpeg_tag{});
+
+    // Convolve the rows and the columns of the image with a fixed kernel
+    gray8_image_t convolved(img.dimensions());
+    gray8_image_t convolved2(img.dimensions());
+
+    // difference kernel
+    float sharpen_1[]={-0.05f,0.0f,0.05f};
+    
+
+    // making resizable kernel object
+    kernel_1d<float> kernel2(sharpen_1,3,1);
+    //compute horizontal edges
+    convolve_rows<gray32f_pixel_t>(color_converted_view<gray8_pixel_t>(const_view(img)),kernel2,view(convolved));
+    //compute vertical edges
+    convolve_cols<gray32f_pixel_t>(color_converted_view<gray8_pixel_t>(const_view(img)),kernel2,view(convolved2));
+    //save horizontal edges
```

Comment:
`save...` just repeates what the self-descriptive `write_view` indicates.
Such comments repeating the actual code are redundant.